File: pingoygou-parent/pinyougou-sellergoods-service-interface/src/main/java/com/pinyougou/service/BrandService.java
package com.pinyougou.service;

import com.pinyougou.pojo.TbBrand;
import com.pinyougou.viewEntity.PageResult;

import java.util.List;
import java.util.Map;

public interface BrandService {
    /**
     * 查找所有
     * @return
     */
    List<TbBrand> findAll();

    /**
     *  分页查找
     * @param page  第几页
     * @param pagesize 一页显示的数量
     * @return
     */
    PageResult findByPage(Integer page,Integer pagesize);

    /**
     *   添加
     * @param brand
     * @return
     */
    int add(TbBrand brand);

    /**
     *  修改
     * @param brand
     * @return
     */
    int updateBrand(TbBrand brand);

    /**
     *  通过id查找
     * @param id
     * @return
     */
    TbBrand findById(Integer id);

    /**
     *  批量删除
     * @param ids
     * @return
     */
    int deletes(Long[] ids);

    /**
     *  查询
     * @param brand
     * @param page
     * @param pagesize
     * @return
     */
    PageResult search(TbBrand brand,Integer page,Integer pagesize);

    /**
     *  查询模板的下拉框选项
     * @return
     */
    List<Map> selectOptionList();
}

File: pingoygou-parent/pinyougou-manager-web/src/main/resources/resources/js/service/brandService.js
//brand的服务层实现
app.service("brandService",function ($http) {
    this.findall=function () {
        return $http.get("/brand/findAll.do");
    }
    this.findPage=function (pagenum,size) {
        return $http.get("/brand/findPage.do?page="+pagenum+"&size="+size);
    }
    this.deletes=function (ids) {
        return $http.get("/brand/deletes.do?ids="+ids);
    }
    this.search=function (pagenum,size,searchEntity) {
        return $http.post("/brand/search.do?page="+pagenum+"&size="+size,searchEntity);
    }
    this.add=function (entity) {
        return $http.post("/brand/add.do",entity);
    }
    this.update=function (entity) {
        return $http.post("/brand/update.do",entity);
    }
    this.findOne=function (id) {
        return $http.get("/brand/findOne.do?id="+id);
    }
});